beamer/config/commands.py

```diff
@@ -131,3 +138,250 @@ def read(
+    if current_fees != desired_fees:
+        yield "RequestManager", "updateFees", *desired_fees
+
+
+def _generate_chain_updates(
+    current_config: Configuration, desired_config: DesiredConfiguration
+) -> Generator[tuple, None, None]:
+    current_chains = current_config.request_manager.chains
+    desired_chains = desired_config.request_manager.chains
+
+    # We don't support chain removals yet.
+    assert set(current_chains).issubset(desired_chains)
+
+    for chain_id, desired_chain in desired_chains.items():
+        current_chain = current_chains.get(chain_id)
+        if current_chain is None or current_chain != desired_chain:
+            yield (
+                "RequestManager",
+                "updateChain",
+                chain_id,
+                desired_chain.finality_period,
+                desired_chain.transfer_cost,
+                desired_chain.target_weight_ppm,
+            )
+
+
+def _generate_token_updates(
+    current_config: Configuration, desired_config: DesiredConfiguration
+) -> Generator[tuple, None, None]:
+    current_tokens = current_config.request_manager.tokens
+    desired_tokens = desired_config.request_manager.tokens
+
+    # First check for removed tokens.
+    for symbol in current_tokens:
+        if symbol not in desired_tokens:
+            address = current_config.token_addresses[symbol]
+            # The token is disabled by setting its transfer limit to zero,
+            # but we set also eth_in_token to zero.
+            yield "RequestManager", "updateToken", address, 0, 0
+
+    # Then check for newly added and modified tokens.
+    for symbol, desired_token in desired_tokens.items():
+        current_token = current_tokens.get(symbol)
+        if current_token is None or current_token != desired_token:
+            address = desired_config.token_addresses[symbol]
+            yield (
+                "RequestManager",
+                "updateToken",
+                address,
+                desired_token.transfer_limit,
+                desired_token.eth_in_token,
+            )
+
+
+def _generate_whitelist_updates(
+    current_config: Configuration, desired_config: DesiredConfiguration
+) -> Generator[tuple, None, None]:
+    def helper(
+        contract: str,
+        current_whitelist: set[ChecksumAddress],
+        desired_whitelist: set[ChecksumAddress],
+    ) -> Generator[tuple, None, None]:
+        for address in current_whitelist - desired_whitelist:
+            yield contract, "removeAllowedLp", address
+
+        for address in desired_whitelist - current_whitelist:
+            yield contract, "addAllowedLp", address
+
+    yield from helper(
+        "RequestManager",
+        current_config.request_manager.whitelist,
+        desired_config.request_manager.whitelist,
+    )
+    yield from helper(
+        "FillManager", current_config.fill_manager.whitelist, desired_config.fill_manager.whitelist
```

Comment:
we may also use same formatting for RequestManager

Reply:
:D like this 
`
yield from helper(
        "FillManager",
        current_config.fill_manager.whitelist,
        desired_config.fill_manager.whitelist,
    )
`

It looks just weird like this :D but ok